### ISSUE TITLE
Configure `io1` volumes for servers

### ISSUE THREAD

**Issue by indytechcook** [department-of-veterans-affairs/va.gov-cms#1159]
Configure the new `io1` volume config available in BRD that was recently added. Currently it defaults to the slower `gp2` SSDs.

**Comment by ElijahLynn** [comment]
Upon thinking further, I think this is going to need to be an EFS volume that we mount on PROD, just like our `/files` mount so it persists between deploys. 

The main instance though, now has `io1` support and we should definitely upgrade all instances to that > ansible/deployment/config/cms-vagov-dev.yml. It will improve opcache rebuild times and other disk based stat operations as well as `tar` operations.

**Comment by indytechcook** [comment]
In my discussions with @ElijahLynn  this isn't needed any longer.

**Comment by ElijahLynn** [comment]
@cmaeng We need to put this back into the backlog as we still want this but just not for the CMS Export System. It has value still so we still want to roll this out soon. The value is around all IO operations and will make every page load a bit faster.

**Comment by indytechcook** [comment]
@ElijahLynn I have. WIP branch: https://github.com/department-of-veterans-affairs/devops/tree/nh-cms-iops.

What value should we set for `provisioned_throughput_in_mibps`:  See https://www.terraform.io/docs/providers/aws/r/efs_file_system.html#provisioned_throughput_in_mibps

**Comment by ElijahLynn** [comment]
PR is here https://github.com/department-of-veterans-affairs/devops/pull/6572

**Comment by ElijahLynn** [comment]
Here are some TAR comparisons, big win on the initial creation w/IO1 at 1.7s vs GP2 at 7.2s.

Source directory - 959MB, 7,246 files:

**IO1 @ 3K IOPS**
`tar --create --file files.tar files-io1/` - **_1.7s_**
`tar --update --file files.tar files-io1/` - **_.2s_**

**GP2 @ 3K IOPS (burst)**
`tar --create --file files.tar files-io1/` - **_7.2s_**
`tar --update --file files.tar files-io1/` - **_.2s_**

![image](https://user-images.githubusercontent.com/1504756/77982758-f0a7ba00-72c1-11ea-857b-d37dd4d1cccc.png)

**Comment by ElijahLynn** [comment]
Application browser loads seem to be unaffected by normal tasks, the IO1 actually seems marginally slower if anything, but that may be do to the proxy connection.

![image](https://user-images.githubusercontent.com/1504756/77983340-a0c9f280-72c3-11ea-9eb9-a0d4d95055f1.png)

So I wrote a loop to query the server from itself for response times to /admin/content and times were identical.

![image](https://user-images.githubusercontent.com/1504756/77983405-d66edb80-72c3-11ea-84f8-9e2ef16bff9f.png)

**IO1**
0.394320
0.154863
0.154933
0.162108
0.171857
0.158230
0.156702
0.153634
0.154012
0.158425
0.154076
0.157786
0.163152
0.156621

**GP2**
0.409567
0.160269
0.158124
0.157763
0.151602
0.155861
0.150735
0.154865
0.154922
0.155016
0.152714
0.145445
0.155006
0.158581
0.153651
0.151664
0.171473

This is all explained by the fact that PHP Opcache is in-memory and doesn't touch much disk when hot. So the IO1 will mainly be of value with TAR operations which we are using for the CMS Content Export System (CES) to make that fast.

**Comment by ElijahLynn** [comment]
Price comparisons of IO1 @ 3K IOPS vs GP2 @ 3K IOPS (burst). Since the price difference is notable, $200/month vs $10/month it might be better to leave the root volume as GP2 and only the files mount as IO1 and test that too.
![image](https://user-images.githubusercontent.com/1504756/77984103-86911400-72c5-11ea-85ff-6ff17a515029.png)

**Comment by ElijahLynn** [comment]
Very interesting results, if `gp2` is used with `tar --create` _and_ the file already exists, it takes ~7 seconds while `io1` takes just 1.7s. HOWEVER, if I manually `rm` the file first then `tar --create`, then `gp2` and `io1` both perform almost identically. `tar --create` will delete the file if it exists and it appears that somehow this deletion takes the extra 5-6s, which seems like a bug in `tar` but likely something I am overlooking. 

So, in conclusion, I think we actually _should_ not merge this and attempt to use the GP2 for now but make sure to use the `rm` technique above and if we want to squeeze out some extra performance gains we can still do so with IO1 and a higher IOPS than 3K. 

![image](https://user-images.githubusercontent.com/1504756/77984967-f3a5a900-72c7-11ea-92a3-e6b5e9067b48.png)
